cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/ControlFlowSensitiveDFGPass.kt

```diff
@@ -498,6 +439,101 @@ open class ControlFlowSensitiveDFGPass(ctx: TranslationContext) : EOGStarterPass
         return state
     }
 
+    protected fun handlePartialAccessExpression(
+        currentNode: Expression,
+        base: Expression,
+        subElement: Node?,
+        doubleState: DFGPassState<Set<Node>>,
+    ) {
+        when (currentNode.access) {
+            AccessValues.WRITE -> {
+                // already set in DFG pass, because otherwise we cannot set the field property
+                // state.push(currentNode.base, PowersetLattice(identitySetOf(currentNode)))
+                val writtenDeclaration = (base as? Reference)?.refersTo
+
+                if (writtenDeclaration != null) {
+                    // we also want to set the last write to our base here.
+                    doubleState.declarationsState[writtenDeclaration] =
+                        PowersetLattice(identitySetOf(base))
+
+                    // Update the state identifier of this node, so that the data flows to later
+                    // member
+                    // expressions accessing the same object/field combination.
+                    doubleState.declarationsState[currentNode.objectIdentifier()] =
+                        PowersetLattice(identitySetOf(currentNode))
+                }
+            }
+
+            AccessValues.READ -> {
+                val writtenDeclaration = (base as? Reference)?.refersTo
+
+                if (writtenDeclaration != null && subElement != null) {
+                    // We do an ugly hack here: We store a (unique) hash out of field declaration
+                    // and
```

Comment:
The comment line breaks are off, probably due to moving it. please double check.